src/errors.rs

```diff
@@ -1,52 +1,23 @@
-use std::error;
-
 use bincode::ErrorKind;
 use thiserror::Error;
 
 #[derive(Debug, Error)]
-pub enum NetlinkError {
-    #[error("Failed to create socket: {0}")]
-    Socket(#[from] Box<dyn error::Error>),
-
-    #[error("Failed to send message: {0}")]
-    Send(String),
-
-    #[error("Netlink error: {0}")]
-    Netlink(String),
-
-    #[error("Failed to decode netlink message: {0}")]
-    NetlinkDecode(String),
-}
-
-#[derive(Debug, Error)]
-pub enum LinkError {
-    #[error("rust-netlink error: {0}")]
-    Netlink(#[from] NetlinkError),
-
+pub enum Error {
```

Comment:
I still think you only need a single variant here -- the `Parse(String)` variant. These are all parse errors and parse errors are not recoverable. Encoding the different underlying parse errors into the type system does not afford the user much - it might as well be a verbose string error to keep things simple.